glib/src/subclass/types.rs

```diff
@@ -233,18 +234,18 @@ impl TypeData {
         self.parent_class
     }
 
-    /// Returns a pointer to the interface implementation specific data.
+    /// Returns a pointer to the class implementation specific data.
     ///
-    /// This is used for interface implementations to store additional data.
-    pub fn get_interface_data(&self, type_: ffi::GType) -> ffi::gpointer {
+    /// This is used for class implementations to store additional data.
+    pub fn get_class_data(&self, type_: ffi::GType) -> ffi::gpointer {
         unsafe {
-            if self.interface_data.is_null() {
+            if self.class_data.is_null() {
                 return ptr::null_mut();
             }
 
-            for &(t, p) in &(*self.interface_data) {
-                if t == type_ {
-                    return p;
+            for (t, p) in &(*self.class_data) {
```

Comment:
How about

```suggestion
            (*self.class_data).get(&type_).unwrap_or(ptr::null_mut())
```

Reply:
Ah yes, that's definitely better than what I did.

Reply:
Although, I wonder if we should have a helper to add something to class_data the same way we have a getter?

Reply:
I think that would be very useful, yes. A setter that allocates the hashmap if not existing, and that then sets the value (and panics if a value was already set!). That setter must be unsafe and must only be called during type initialization.

You could also add a function like the `get()` above that is unsafe and `get_mut()`.

Aaaaand. You can also make the whole API slightly more safe and less ugly at no meaningful cost at runtime after initialization by having it store `Box<dyn Any + Send + Sync>` instead of working with raw pointers for the hashmap values.

Reply:
> I think that would be very useful, yes. A setter that allocates the hashmap if not existing, and that then sets the value (and panics if a value was already set!). That setter must be unsafe and must only be called during type initialization.

Done, I'm not sure how can I ensure it's only called at type initialization ? Or that should be just in the safety section of the function documentation ?

> 
> You could also add a function like the `get()` above that is unsafe and `get_mut()`.

What would be the difference between get() & get_class_data? make the former return a specific type instead of a pointer?

Reply:
> Done, I'm not sure how can I ensure it's only called at type initialization ? Or that should be just in the safety section of the function documentation ?

Yes in the docs, and that's why it's unsafe :)

> What would be the difference between get() & get_class_data? make the former return a specific type instead of a pointer?

I was just lazy to type. I meant `fn get_class_data() -> &dyn Any + Send + Sync` and `unsafe fn get_class_data_mut() -> &mut dyn Any + Send + Sync`

Reply:
Thanks